powertools-metrics/src/main/java/software/amazon/lambda/powertools/metrics/internal/EmfMetricsLogger.java

```diff
@@ -221,43 +240,22 @@ public void captureColdStartMetric(software.amazon.lambda.powertools.metrics.mod
     }
 
     @Override
-    public void flushSingleMetric(String name, double value, MetricUnit unit, String namespace,
-            software.amazon.lambda.powertools.metrics.model.DimensionSet dimensions) {
+    public void flushMetrics(Consumer<Metrics> metricsConsumer) {
         if (isMetricsDisabled()) {
             LOGGER.debug("Metrics are disabled, skipping single metric flush");
             return;
         }
-
-        Validator.validateNamespace(namespace);
-
-        // Create a new logger for this single metric
-        software.amazon.cloudwatchlogs.emf.logger.MetricsLogger singleMetricLogger = new software.amazon.cloudwatchlogs.emf.logger.MetricsLogger(
-                environmentProvider);
-
-        try {
-            singleMetricLogger.setNamespace(namespace);
-        } catch (Exception e) {
-            LOGGER.error("Namespace cannot be set for single metric due to an error in EMF", e);
+        // Create a new instance, inheriting namespace/dimensions state
+        EmfMetricsLogger metrics = new EmfMetricsLogger(environmentProvider, metricsContext);
+        if (namespace != null) {
+            metrics.setNamespace(this.namespace);
         }
+        defaultDimensions.forEach(metrics::addDimension);
+        metadata.forEach(metrics::addMetadata);
```

Comment:
We haven't decided yet in the issue if we want to auto-inherit the default metrics logger configuration. I see in v1 this is the behavior we have.